Imager.js

```diff
@@ -142,30 +143,36 @@
           });
         }
 
-        if (elements) {
-            this.divs = applyEach(elements, returnFn);
-            this.selector = null;
-        }
-        else {
-            this.divs = applyEach(doc.querySelectorAll(this.selector), returnFn);
-        }
-
+        this.divs = [];
+        this.add(elements || this.selector);
         this.ready(opts.onReady);
-        this.changeDivsToEmptyImages(this.divs);
 
-        setTimeout(function(){
+        setTimeout(function () {
             self.init();
         }, 0);
-    }
+    };
+
+    Imager.prototype.add = function (elementsOrSelector) {
+
+        elementsOrSelector = elementsOrSelector || this.selector;
```

Comment:
@WebSeed I see what you mean. Here are my two concerns:

- a one line setup is nice and implicit values can be acceptable at that stage – having a mutating function without parameters seems odd
- a call to `.add()` will rematch and add existing responsive images in the pool – if you call it 5 times without adding any new image, the `imgr.divs` length will be multiplied by 5 anyway

If elements are loaded in the DOM, *you* have a reference to those elements, and they should be explicitly passed as an argument of `add` instead of *expecting* them to be matched by the initial selector.

It is a more functional-ish approach, more rigorous but less pragmatic though.

tl;dr: an empty call to `.add()` seems to me to be rather an `.update()` method.